src/jit/morph.cpp

```diff
@@ -2599,7 +2602,7 @@ GenTree* Compiler::fgInsertCommaFormTemp(GenTree** ppTree, CORINFO_CLASS_HANDLE
 //    This method only computes the arg table and arg entries for the call (the fgArgInfo),
 //    and makes no modification of the args themselves.
 //
-void Compiler::fgInitArgInfo(GenTreeCall* call)
+void Compiler::fgInitArgInfo(GenTreeCall* call, bool reInitArgInfo/*=false*/)
```

Comment:
Given how specific this case is it would make more sense to have the caller clear `fgArgInfo` instead of modifying this function's signature.

Reply:
My hope would be for the function to abstract that away. More importantly, I hope that eventually we can quickly check if there have been changes to the args of the caller if the info has been populated already and only create new arg entries for the difference. It seems unnecessary and costly long term to just blow away the old data.

Reply:
> More importantly, I hope that eventually we can quickly check if there have been changes to the args of the caller if the info has been populated already and only create new arg entries for the difference.

This function (and pretty much everything related to calls) is already horrible as it is. Piling more logic on top of all this is just going to make it worse. Especially considering that argument changes that also require arg info changes aren't exactly common.

Reply:
Addressed

Reply:
I'm OK with this resolution, but as we continue to clean up and unify information about both incoming and outgoing args, I think it would be desirable to avoid having to make these late decisions that could potentially lead to confusion down the line.